### PR TITLE
Add error callback for showStep (Issue #274)

### PR DESCRIPTION
**Pull request by travstone**
Calling showStep without a proper step target fails, but the tour error
callback is not invoked. This change fixes that, and sets the current
step number to the step that failed for debugging purposes

I'm getting a different error in my codepen example (used in the issue report) related to the 'bubble_default' template, so I wasn't able to test this. I do have this fix in a client code-base however, and that is working fine.

This may be imperfect, if so I apologize. Just thought I'd try to get this in for the maintenance release if possible.

### CODE REVIEW

**Comment by zimmi88** [comment]
1 test failed. =(

Lemme take a look and see what it's complaining about...

File: src/js/hopscotch.js [hunk 1]
```diff
@@ -1945,6 +1945,8 @@
     this.showStep = function(stepNum) {
       var step = currTour.steps[stepNum];
       if(!utils.getStepTarget(step)) {
+        currStepNum = stepNum;
```
Comment (zimmi88) [inline src/js/hopscotch.js]: Let's remove this. IMO, if the step's target doesn't exist, we should leave the tour where it is. This also is probably the source of the build failure.

**Comment by travstone** [comment]
Gah! Sorry for the bad PR. Can definitely remove that bit, but will be slightly less clear if the callback is returning the failed stepNum. Not a deal-breaker though.

Shall I refactor and resubmit? I'm not sure what the protocol is here. Thanks for taking a look!

**Comment by zimmi88** [comment]
Hmm, fair point. So, maybe it would make sense to optimistically set `currStepNum`, then revert `currStepNum` back on failure (after calling `invokeEventCallbacks()`). This behavior is consistent with `goToStepWithTarget()`, except in that case we keep stepping forward until we find a valid step (which is not intended when we call `showStep` at present).

You can leave this PR open. Go ahead and push a new commit to the feature branch you have open at present. The commits on a PR branch will get squashed when they're merged to master.

File: src/js/hopscotch.js [hunk 2]
```diff
@@ -2430,8 +2431,90 @@
 // Template includes, placed inside a closure to ensure we don't
 // end up declaring our shim globally.
 (function(){
-// @@include('../../src/tl/_template_headers.js') //
-// @@include('../../tmp/js/hopscotch_templates.js') //
+var _ = {};
```
Comment (zimmi88) [inline src/js/hopscotch.js]: This shouldn't be included in /src. This code gets added in when running `grunt`. If you need the output from `grunt` for testing purposes, the built files will be placed in the `tmp/` folder.

**Comment by travstone** [comment]
Yep, sorry about that- still getting used to some of this. Am trying to test now, had to install node etc. but now I understand why I was getting the error in my codepen example.

**Comment by travstone** [comment]
Ok, checked this back in here; all tests passing for me locally, and my codepen example is working correctly. Thanks!

File: src/js/hopscotch.js [hunk 3]
```diff
@@ -1945,6 +1945,10 @@
     this.showStep = function(stepNum) {
       var step = currTour.steps[stepNum];
       if(!utils.getStepTarget(step)) {
+        var prevStepNum = currStepNum;
```
Comment (zimmi88) [inline src/js/hopscotch.js]: Nitpick... hoist the var declaration to the top of the function.

**Comment by zimmi88** [comment]
Cool, this PR is looking good to me. Just commented with one minor nitpick... once that's fixed, I'll mark this as Ready to Merge. I'll pull this in tonight or tomorrow. Thanks!

**Comment by travstone** [comment]
No problem! I should have thought of that; too focused on making sure I built it properly to observe good code style. Thanks for catching that.

**Pull request merged**